cmd/ako-main/main.go

```diff
@@ -335,3 +359,41 @@ func init() {
 	flag.StringVar(&kubeconfig, "kubeconfig", def_kube_config, "Path to a kubeconfig. Only required if out-of-cluster.")
 	flag.StringVar(&masterURL, "master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
 }
+
+func istioWatcherEvents(watcher *fsnotify.Watcher, kc *kubernetes.Clientset, istioUpdateCh *chan struct{}) {
+	for {
+		select {
+		case event, ok := <-watcher.Events:
+			if !ok {
+				return
+			}
+			if event.Op == fsnotify.Write || event.Op == fsnotify.Create {
+				utils.AviLog.Infof("Istio watcher event, modified file: %s", event.Name)
+
+				if strings.HasSuffix(event.Name, "istio-output-certs") {
+					watcher.Remove("/etc")
+					utils.AviLog.Infof("removed path to /etc to istio watcher")
+					watcher.Add(lib.IstioCertOutputPath)
+					utils.AviLog.Infof("added path to %s to istio watcher", lib.IstioCertOutputPath)
+					initIstioSecrets(kc, istioUpdateCh)
+
+				} else {
+					lib.CreateIstioSecretFromCert(event.Name, kc)
+				}
+			}
+
+		case err, ok := <-watcher.Errors:
+			if !ok {
+				return
+			}
+			utils.AviLog.Warnf("Istio watcher event, error: %s", err.Error())
+		}
+	}
+}
+
+func initIstioSecrets(kc *kubernetes.Clientset, istioUpdateCh *chan struct{}) {
+	lib.CreateIstioSecretFromCert(lib.IstioCertOutputPath+"/cert-chain.pem", kc)
+	lib.CreateIstioSecretFromCert(lib.IstioCertOutputPath+"/key.pem", kc)
+	lib.CreateIstioSecretFromCert(lib.IstioCertOutputPath+"/root-cert.pem", kc)
+	close(*istioUpdateCh)
```

Comment:
this might create an issue on the second write to the secret file. closing an already closed channel will result in a panic I believe.

Reply:
had a discussion with Arif and this wont get called twice.